### ISSUE TITLE
trying to control a 57946 controller (supposedly supported)

### ISSUE THREAD

**Issue by GHSTUDIO1** [sebr/bhyve-home-assistant#250]
Many issues

1) I have three switches (for my 3 programs).  All are defined, but all are unavailable.

2) the switch names are switch.zone; switch.zone_1_zone_2; switch.zone_3    (zone 2 is wrong)

3) I can't find any "rain delay" switch

4) I only see one program switch with  none of the attributes shown in the documentation. 

-->

**Comment by sebr** [comment]
You need to add the device diagnostics, as requested in the issue template

**Comment by GHSTUDIO1** [comment]
I have the following in my configuration.yaml file:

 

logger:

   default: critical    

   logs:

    custom_components.bhyve: debug

    pybhyve: debug

 

However, I don’t see any log entries.  

 

I’ve checked and Bhyve is in my custom components folder and I have a (normal) warning message:

 

2024-06-28 07:51:29.095 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration bhyve which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

 

But there are no log entries.  Am I looking in the wrong place???

 

 

BTW, the device integration shows as UPNP……it should show Orbit as it’s name, shouldn’t it???

**Comment by sebr** [comment]
Please attach the Home Assistant diagnostics for the B-hyve device. Sensitive information is redacted when compiling the diagnostic data.

* Home Assistant diagnostics: https://www.home-assistant.io/integrations/diagnostics/
* Diagnostic data implementation: https://github.com/sebr/bhyve-home-assistant/blob/main/custom_components/bhyve/diagnostics.py
* Example of how to generate: https://user-images.githubusercontent.com/81972/179869298-3d2a33ca-f847-4a25-8e77-20757e49f914.png

**Comment by GHSTUDIO1** [comment]
[orbi_debug.txt](https://github.com/user-attachments/files/16044759/orbi_debug.txt)

**Comment by sebr** [comment]
@GHSTUDIO1 I cannot help you if you do not follow instructions.

**Comment by GHSTUDIO1** [comment]
I followed the instructions.  Debug is on and I sent you the log   Is it the wrong log???

**Comment by sebr** [comment]
At this stage I don't need your logs, but rather the device diagnostics. I refer you to my comment here:

https://github.com/sebr/bhyve-home-assistant/issues/250#issuecomment-2198514160

**Comment by GHSTUDIO1** [comment]
1.	I can’t find a diagnostics integration … so I can’t add it.  I have tried looking at system integration, Haacs and addins.  If there is an integration I should have for this, I don’t know how to addit.
2.	Diagnostics,py is installed in the custom components section with your other code.
3.	I don’t have an option to print out any diagnostic data….the only thing I can send is what I sent.

 

I wish I could find the diagnostics integration that is supposedly included in the Home Assistant package.   Note: I added  diagnostics:   to my comfiguration.yaml which didn’t change anything

 

I’m trying to help get this fixed….I must be missing something.

**Comment by sebr** [comment]
Diagnostics is enabled and bundled by default.

Did you look at the screenshot? https://user-images.githubusercontent.com/81972/179869298-3d2a33ca-f847-4a25-8e77-20757e49f914.png

**Comment by GHSTUDIO1** [comment]
close this issue.....the problem was that the integration did not load correctly.  

On my initial install, I entered an incorrect password at least twice and the integration remembered the incorrect password.  Attempting to reinstall, I got a message that the integration had already been registered, but as I said, it was registered with the wrong password so the integration was unable to start. 

The resolution was to delete the bhyve section in  /homeassistant/.storage/core.config_entries  and reinstall the integration.  I might have been able to fix by just replacing the password in core.config_entries, but I chose to just delete that section.
